### PR TITLE
Patron dashboard

### PR DESCRIPTION
**Pull request by Stamates**
Added truncated bio and link to contact page for patron dashboard (grid of staff members).  Also filtered out admin from displayed staff members.

### CODE REVIEW

File: templates/main.html [hunk 1]
```diff
@@ -12,9 +12,13 @@
       <div class="row">
       {% for staff in staff %}
         <div class="col-md-4 col-sm-6">
-          <a href="/profile/{{ staff.username }}">
-          {% include "user_profile.html" %}
-          </a>
+            <a href="/profile/{{ staff.username }}">
```
Comment (hitbox) [inline templates/main.html]: We can use `url_for` here and for all `a href`s too. `<a href="{{ url_for('profile', uname=staff.username) }}">`.

File: library.py [hunk 2]
```diff
@@ -197,5 +202,13 @@ def edit_profile(uname):
             return render_template('profile.html', bio=new_bio)
 
 
+@app.route("/contact/<uname>", methods=['GET', 'POST'])
+def contact(uname):
+    if request.method == "GET":  # regular get, present the form to user to edit.
+        return render_template('contact.html', staff=uname)
+
+    elif request.method == "POST":  # form was submitted, update database
+        return redirect('/profile/' + uname)
```
Comment (hitbox) [inline library.py]: Can make this `return redirect(url_for('profile'), uname)` if you like a bit more "pythonic".

File: templates/main.html (hunk 1 above)
Comment (hitbox) [inline templates/main.html]: Really useful if we ever change the url structure.

**Comment by hitbox** [comment]
Looks good to me. I made a few notes for suggestions. Pretty optional stuff.

File: library.py (hunk 2 above)
Comment (Stamates) [inline library.py]: Yeah... I tried that originally but couldn't figure out the proper structure for how to add the uname reference.... it's actually `return redirect(url_for('profile', uname=uname))`.  I figured it out based on the syntax you used for the href link.  Thanks!

File: templates/main.html (hunk 1 above)
Comment (Stamates) [inline templates/main.html]: Thanks!

**Pull request merged**